tests/unit/unit-walker.cpp

```diff
@@ -37,12 +38,57 @@ class UnitWalkerTests : public ::testing::Test,
 
     bvsort = s->make_sort(BV, 4);
     funsort = s->make_sort(FUNCTION, SortVec{ bvsort, bvsort });
+    fun2sort = s->make_sort(FUNCTION, SortVec{ bvsort, bvsort, bvsort });
     arrsort = s->make_sort(ARRAY, bvsort, bvsort);
+    boolsort = s->make_sort(BOOL);
   }
   SmtSolver s;
-  Sort bvsort, funsort, arrsort;
+  Sort bvsort, funsort, fun2sort, arrsort, boolsort;
 };
 
+/* Custom Tree Walker that builds up a map assigning fresh boolean variables to each occurrence of each term in the formula it visits.
+ * One example of overriding visit_term to customize the behavior of the walker. */
+class IndicatorTreeWalker : public TreeWalker {
```

Comment:
plz specify in which test this class will be used.

Reply:
.